lib/sqlalchemy/orm/session.py

```diff
@@ -1724,6 +1724,31 @@ def scalar(
             **kw
         ).scalar()
 
+    def scalars(
+        self,
+        statement,
+        params=None,
+        execution_options=util.EMPTY_DICT,
+        bind_arguments=None,
+        **kw
+    ):
+        """Execute a statement and return the results as scalars.
+
+        Usage and parameters are the same as that of
+        :meth:`_orm.Session.execute`; the return result is a
+        :class:`_result.ScalarResult` filtering object which
```

Comment:
**Federico Caselli** (CaselIT) wrote:

I'm not an authority on English,  but filter here seems to imply that some rows are filtered out. Maybe it's just me :).
An alternative off the top of my head "yielding objects" ?

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3065

Reply:
I agree that "filtering" can be interpreted as a selection process instead of a conversion, but if we are going to change it then it should be changed in a few more places that currently use this terminology in existing code. For example, [here is one](https://github.com/sqlalchemy/sqlalchemy/blob/680baea6b83c497e862f3f0ade28e11940cfd895/lib/sqlalchemy/engine/result.py#L857).

The word "mapping" seems like it is more appropriate than filtering here, but mapping can also make things confusing in the context of the ORM.

Reply:
I had not noticed it was already used there.



> The word "mapping" seems like it is more appropriate than filtering here, but mapping can also make things confusing in the context of the ORM.

I agree that it would fit, but it may indeed clash with orm mapping.

Maybe a general "transform"?

Reply:
**mike bayer** (zzzeek) wrote:

lets do "transform" in a separate fix since I used "filtering" already.  i believe this might be internal terminology.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3065